lib/screen/onboarding/import_address/select_addresses.dart

```diff
@@ -0,0 +1,307 @@
+import 'dart:io';
+
+import 'package:autonomy_flutter/common/injector.dart';
+import 'package:autonomy_flutter/database/entity/persona.dart';
+import 'package:autonomy_flutter/screen/app_router.dart';
+import 'package:autonomy_flutter/screen/bloc/scan_wallet/scan_wallet_bloc.dart';
+import 'package:autonomy_flutter/screen/bloc/scan_wallet/scan_wallet_state.dart';
+import 'package:autonomy_flutter/service/account_service.dart';
+import 'package:autonomy_flutter/service/configuration_service.dart';
+import 'package:autonomy_flutter/util/constants.dart';
+import 'package:autonomy_flutter/util/string_ext.dart';
+import 'package:autonomy_flutter/util/style.dart';
+import 'package:autonomy_flutter/view/au_toggle.dart';
+import 'package:autonomy_flutter/view/back_appbar.dart';
+import 'package:autonomy_flutter/view/crypto_view.dart';
+import 'package:autonomy_flutter/view/primary_button.dart';
+import 'package:autonomy_flutter/view/radio_check_box.dart';
+import 'package:autonomy_flutter/view/responsive.dart';
+import 'package:autonomy_theme/autonomy_theme.dart';
+import 'package:collection/collection.dart';
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class SelectAddressesPage extends StatefulWidget {
+  static const String tag = 'select_addresses_page';
+  final SelectAddressesPayload payload;
+
+  const SelectAddressesPage({Key? key, required this.payload})
+      : super(key: key);
+
+  @override
+  State<SelectAddressesPage> createState() => _SelectAddressesPageState();
+}
+
+class _SelectAddressesPageState extends State<SelectAddressesPage> {
+  final List<String> _importedAddresses = [];
+
+  int index = 0;
+  final List<AddressInfo> _addresses = [];
+  final List<AddressInfo> _selectedAddresses = [];
+  bool _onlyBalance = true;
+  bool _addingAddresses = false;
+
+  @override
+  void initState() {
+    super.initState();
+    _callBloc();
+    fetchImportedAddresses();
+  }
+
+  Future<void> fetchImportedAddresses() async {
+    final importedAddresses = await widget.payload.persona.getAddresses();
+    setState(() {
+      _importedAddresses.addAll(importedAddresses);
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    return Scaffold(
+      appBar: getBackAppBar(
+        context,
+        title: "import_address".tr(),
+        onBack: () {
+          Navigator.of(context).pop();
+        },
+      ),
+      body: BlocConsumer<ScanWalletBloc, ScanWalletState>(
+          listener: (context, scanState) {
+        if (scanState.addresses.isNotEmpty && !scanState.isScanning) {
+          setState(() {
```

Comment:
Should not use `setState` here. Instead, try to get the value in the builder and bind it to the UI.